File: app/api/routes/members.py
"""
app/api/routes/members.py — /api/members CRUD 엔드포인트
"""

from typing import List

from fastapi import APIRouter, HTTPException, Response

from app.models.member import MemberInput, MemberRecord, MemberUpdate
from app.services.member_service import MemberService

router = APIRouter()


@router.get("", response_model=List[MemberRecord])
async def list_members():
    return await MemberService().get_all()


@router.post("", status_code=201, response_model=MemberRecord)
async def create_member(body: MemberInput):
    try:
        return await MemberService().create(body)
    except Exception as e:
        raise HTTPException(status_code=500, detail=str(e))


@router.patch("/{member_id}", response_model=MemberRecord)
async def update_member(member_id: str, body: MemberUpdate):
    try:
        return await MemberService().update(member_id, body)
    except KeyError as e:
        raise HTTPException(status_code=404, detail=str(e))
    except Exception as e:
        raise HTTPException(status_code=500, detail=str(e))


@router.delete("/{member_id}", status_code=204)
async def delete_member(member_id: str):
    try:
        await MemberService().delete(member_id)
        return Response(status_code=204)
    except KeyError as e:
        raise HTTPException(status_code=404, detail=str(e))
    except Exception as e:
        raise HTTPException(status_code=500, detail=str(e))

File: frontend/src/api/members.ts
import type { MemberInput, MemberRecord, MemberUpdate } from "../types";

const BASE = "/api";

export async function getMembers(): Promise<MemberRecord[]> {
  const res = await fetch(`${BASE}/members`);
  if (!res.ok) throw new Error(`HTTP ${res.status}`);
  return res.json();
}

export async function createMember(data: MemberInput): Promise<MemberRecord> {
  const res = await fetch(`${BASE}/members`, {
    method: "POST",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify(data),
  });
  if (!res.ok) {
    const err = await res.json().catch(() => ({}));
    throw new Error(err.detail ?? `HTTP ${res.status}`);
  }
  return res.json();
}

export async function updateMember(id: string, data: MemberUpdate): Promise<MemberRecord> {
  const res = await fetch(`${BASE}/members/${id}`, {
    method: "PATCH",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify(data),
  });
  if (!res.ok) {
    const err = await res.json().catch(() => ({}));
    throw new Error(err.detail ?? `HTTP ${res.status}`);
  }
  return res.json();
}

export async function deleteMember(id: string): Promise<void> {
  const res = await fetch(`${BASE}/members/${id}`, { method: "DELETE" });
  if (!res.ok) {
    const err = await res.json().catch(() => ({}));
    throw new Error(err.detail ?? `HTTP ${res.status}`);
  }
}
